src/routes/safe/components/Apps/utils.js

```diff
@@ -15,22 +18,44 @@ export const getAppInfoFromOrigin = (origin: string) => {
 }
 
 export const getAppInfoFromUrl = async (appUrl: string) => {
+  let res = { id: undefined, url: appUrl, name: 'unknown', iconUrl: appsIconSvg, error: true }
+
   try {
     const appInfo = await axios.get(`${appUrl}/manifest.json`)
-    const res = { url: appUrl, ...appInfo.data, iconUrl: appsIconSvg }
+
+    // verify imported app fulfil safe requirements
+    if (
+      !appInfo ||
```

Comment:
Please do a proper error message here .. I am currently trying to get this to work for an example and it is a pain to debug what information is missing or what the error is